_includes/repl.html

```diff
@@ -89,7 +90,73 @@
 <script src="assets/js/luau_mode.js"></script>
 <!-- CodeMirror Luau Editor (MUST BE LOADED AFTER CODEMIRROR!) -->
 <script>
+    var MAX_QUERY_PARAM_LENGTH = 0x7FF;
+    function getValueFromQueryString(key) {
+        return new URLSearchParams(window.location.search).get(key);
+    }
+    async function compressString(str) {
+        if (!TextEncoder || !CompressionStream || !Response || !Blob) return btoa(str);
+        console.info('Using gzip compression');
+        console.info('Original string length:', str.length);
+        console.info('base64encoded string length:', btoa(str).length);
+        var encoder = new TextEncoder();
+        var uint8Array = encoder.encode(str);
+
+        var compressedStream = new Response(
+            new Blob([uint8Array]).stream().pipeThrough(new CompressionStream('gzip'))
+        ).arrayBuffer();
+
+        var compressedArrayBuffer = await compressedStream;
+        var compressedUint8Array = new Uint8Array(compressedArrayBuffer);
+
+        var compressedString = btoa(String.fromCharCode(...compressedUint8Array));
```

Comment:
Didn't know JS got spreads!

Reply:
yeah it's had spreads since ES6, so it should be pretty well supported by this point. Same goes for the compression API (which has a fallback)